apps/server/src/shared/domain/rules/team.rule.spec.ts

```diff
@@ -33,35 +30,65 @@ describe('TeamRule', () => {
 		await orm.close();
 	});
 
-	beforeEach(() => {
-		role = roleFactory.build({ permissions: [permissionA] });
-		teamRole = roleFactory.build({ permissions: [permissionB] });
-		user = userFactory.build({ roles: [role] });
-		entity = teamFactory.withRoleAndUserId(teamRole, user.id).build();
-	});
+	const setup = () => {
+		const role = roleFactory.buildWithId({ permissions: [permissionA] });
+		const teamRole1 = roleFactory.buildWithId({ permissions: [teamPermissionA] });
+		const teamRole2 = roleFactory.buildWithId({ permissions: [teamPermissionB], roles: [teamRole1] });
+		const teamRole = roleFactory.buildWithId({ permissions: [teamPermissionC], roles: [teamRole2] });
+		const user = userFactory.buildWithId({ roles: [role] });
+		const entity = teamFactory.withRoleAndUserId(teamRole, user.id).build();
+		return {
+			role,
+			teamRole1,
+			teamRole2,
+			teamRole,
+			user,
+			entity,
+		};
+	};
 
 	describe('isApplicable', () => {
 		it('should return truthy', () => {
-			expect(() => service.isApplicable(entity.teamUsers[0].user, entity)).toBeTruthy();
+			const { user, entity } = setup();
+			expect(() => service.isApplicable(user, entity)).toBeTruthy();
 		});
 	});
 
 	describe('hasPermission', () => {
-		it('should return "true" if user in scope', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionB]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionA]));
 			expect(res).toBe(true);
 		});
 
-		it('should return "false" if user has not permission', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionC]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionB]));
+			expect(res).toBe(true);
+		});
+
+		it('should return "true" if user in team scope', () => {
```

Comment:
There are three tests with the same name. If they test different things, they should have different names.

Reply:
I think what you actually test here is if `this.utils.hasAllPermissionsByRole`can handle inner roles. But this should be tested in the dedicated tests of this method.

I would say you should mock this method here but I catched some discussion about not mocking sync functions in the last days. Maybe you could brief me on that... 

Anyway, one test case should be enough here, also one teamRole in the setup.

Reply:
Thats related to the missing describe blocks. The name of the tests will stay the same than.
```ts
describe('WHEN user is in team scope', () => {
    it('should return "true"', () => {})
}
```

Reply:
There is the decision by arc chapter that rules should be tested with an integration approach. So implementation is correct here.